src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -113,6 +122,21 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
+            if (this.ParameterSetName.Equals(ArmParameterSet, StringComparison.Ordinal))
+            {
+                MetadataResponse metadataEndpoints = null;
+                metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
+                string domain = EnvironmentHelper.RetrieveDomain(metadataEndpoints.PortalEndpoint);
+                ActiveDirectoryEndpoint = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/') + '/';
```

Comment:
should check metadataendpoints.authentication is not null as well as metadataendpoints.authentication.LoginEndpoint